src/adc.rs

```diff
@@ -317,46 +395,37 @@ macro_rules! adc_hal {
                 pub fn $adcX(adc: $ADC, delay: &mut Delay,
                              prec: rec::$Rec, clocks: &CoreClocks
                 ) -> Self {
-                    let mut s = Self {
-                        rb: adc,
-                        sample_time: AdcSampleTime::default(),
-                        resolution: Resolution::SIXTEENBIT,
-                        lshift: AdcLshift::default(),
-                        _enabled: PhantomData,
-                    };
+                    let mut a = Self::default_from_rb(adc);
```

Comment:
Would recommend a rename from `a` -> `adc` (or something similar. Benefit of rebinding the `adc` is that we consume it here as well.)

Reply:
Agreed, thanks!